discord/http.py

```diff
@@ -162,33 +137,35 @@ class HTTPClient:
 
     def __init__(
         self,
+        bucket_storage: BucketStorageProtocol,
```

Comment:
Can we turn this into an optional keyword-only argument in order to prevent a breaking change?